proxy/src/main/java/io/openlineage/proxy/api/models/LineageStream.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.openlineage.proxy.api.models;
+
+
+/**
+ * LineageStream provides the generic implementation of the backend destinations supported by the proxy backend.
+ */
+public abstract class LineageStream
+{
+  /**
+   * The Type enum (and JsonSubTypes above) are extended for each new type of destination that the proxy backend supports.
+   * There is a subtype class for each of these destination types.
+   */
+  enum Type
+  {
+    CONSOLE,
+    KAFKA,
+    KINESIS
```

Comment:
I would remove the KINESIS until it gets implemented.

Reply:
Yeah, good point +1 (Was initially added to illustrate multiple writes to lineage streams).